### PR TITLE
[SUPPORT] : Add Redux

### PR DESCRIPTION
**Pull request by mcayuelas-ledger**
### 📝 Description

Init Redux Toolkit 

### ❓ Context

- **Linked resource(s)**: `` 

### 🚀 Expectations to reach

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

**Review by sshmaxime — DISMISSED** [review]
lgtm ! Maybe i'd add a components that fetches and update the store and add it to the main page

**Comment by mcayuelas-ledger** [comment]
Adding some Docs !

File: tools/test.tools.tsx [hunk 1]
```diff
@@ -2,12 +2,18 @@ import React, { ReactElement } from "react";
...
+import { appReducer } from "@/redux/store";
 
 type PropsTheme = {
   children: React.ReactNode;
   theme?: "dark" | "light";
 };
 
+const mockedStore = configureStore({
+  reducer: appReducer,
+});
```
Comment (KVNLS) [inline tools/test.tools.tsx:16]: You might not do it above the custom render function as here you will define a store that will be shared by all tests and you can have side effects.
Please have a look at the documentation:
You can initiate the store directly as a default value or inside the function. That way you will isolate each test
https://redux.js.org/usage/writing-tests#setting-up-a-reusable-test-render-function
Moreover, we will want to be able to set the store in a specific state: https://redux.js.org/usage/writing-tests#preparing-initial-test-state

File: tools/test.tools.tsx [hunk 2]
```diff
@@ -16,9 +22,11 @@ type PropsTheme = {
  */
 const AllProviders = ({ children, theme = "dark" }: PropsTheme) => {
   return (
-    <StyleProvider selectedPalette={theme} fontsPath="/fonts">
-      {children}
-    </StyleProvider>
+    <Provider store={mockedStore}>
```
Comment (KVNLS) [inline tools/test.tools.tsx:25]: You can rename `mockedStore` as `store` because it's not a mock

**Review by KVNLS — CHANGES_REQUESTED** [review]

**Review by KVNLS — APPROVED** [review]
Looks great!

**Pull request merged**